refactor: Various PHP code modernizations

aragon999 opened the pull request:
### 1. Why is this change necessary?
Just several things which accumulated over time, making the code more modern (and in my opinion more readable).

### 2. What does this change do, exactly?
Rewrite some PHP code in a more modern style.

### 3. Describe each step to reproduce the issue or behaviour.
:eyes: 

### 4. Please link to the relevant issues (if any).
\-

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/shopware/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfilled them

aragon999 committed:
refactor: Various PHP code modernizations

aragon999 committed:
fixup: phpstan errors

aragon999 commented:
Note I also just pushed a commit to fix the missing iterable type errors in the "touched" files, however I am not sure how and where to add the domain exception in the `CreateHydratorCommand` (and if it is even needed at that location).

mitelg commented:
you can add methods to `\Shopware\Core\Framework\DataAbstractionLayer\DataAbstractionLayerException`

and as you are on it, it would be nice if you introduce the Symfony Filesystem component to that command, instead of using native functions :innocent:

aragon999 commented:
@mitelg maybe the check that all feature flags need to be enabled, can be deleted? Or do you (or someone else) know why this is needed?

mitelg commented:
I have not clue :sweat_smile: can probably be removed completely :+1:

aragon999 committed:
fixup: Remove feature flag error and replace by warning

aragon999 committed:
fixup: Refactor replace filesystem by Symfony abstraction

aragon999 commented:
Okay, I replaced the error by a warning, it might be related to the fact that if some field are behind a feature flag, that they are skipped.

Furthermore I replaced the native file system calls.

aragon999 committed:
fixup: Replace catched exceptions

mitelg commented:
thanks for your contribution @aragon999 👍🎉💙
